amethyst_network/src/network_socket.rs

```diff
@@ -28,6 +30,16 @@ struct RawEvent {
     pub source: SocketAddr,
 }
 
+impl From<Packet> for RawEvent {
+    fn from(packet: Packet) -> Self {
+        RawEvent {
+            byte_count: packet.payload().len(),
+            data: packet.payload().to_owned(),
```

Comment:
I think you can drop this `to_owned` now? you should be able to move the payload out.

Reply:
I am not sure but you move ownership of `Packet` to this function I think you won't be able to use it afterward at all.